app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -73,6 +73,19 @@ def puppet_master(f)
                :help_inline => _("Use this puppet server as an initial Puppet Server or to execute puppet runs") }
   end
 
+  def realm_field(f)
+    # Don't show this if we have no Realms, otherwise always include blank
+    # so the user can choose not to use a Realm on this host
+    return if Realm.count == 0
+    return unless SETTINGS[:unattended] && @host.managed
```

Comment:
I'd check ```SETTINGS[:unattended] == true```, just to protect this from being true in the event it's ```falsw``` or some other typo.